File: ios/Ezber/Models/Reciter.swift
import Foundation

/// A reciter whose audio can be streamed or downloaded. The real content
/// pipeline will fill these from the content store; placeholder values are used
/// until it lands.
struct Reciter: Identifiable, Hashable, Codable {
    let id: String
    let name: String
    let style: String
    let languageName: String
    let sampleURL: URL?
    let audioBaseURL: URL?
    let license: String?

    init(
        id: String,
        name: String,
        style: String,
        languageName: String,
        sampleURL: URL? = nil,
        audioBaseURL: URL? = nil,
        license: String? = nil
    ) {
        self.id = id
        self.name = name
        self.style = style
        self.languageName = languageName
        self.sampleURL = sampleURL
        self.audioBaseURL = audioBaseURL
        self.license = license
    }
}

File: android/app/src/main/java/app/ezber/android/models/Reciter.kt
package app.ezber.android.models

/**
 * A reciter whose audio can be streamed or downloaded. The content pipeline
 * fills these from the `reciters` table; `enabled = false` entries are still
 * shown so the rights state stays visible.
 */
data class Reciter(
    val id: Int,
    val remoteId: String,
    val name: String,
    val style: String? = null,
    val qirat: String? = null,
    val attribution: String = "",
    val licenseId: String = "",
    val hasSegments: Boolean = false,
    val enabled: Boolean = true,
    val source: String = "",
    val licenseUrl: String = "",
    val licenseEvidenceUrl: String = "",
) {
    val subtitle: String get() = listOfNotNull(style, qirat).filter { it.isNotBlank() }.joinToString(" · ")
}
